client/src/modules/StudentDashboard/components/TasksStatsCard.tsx

```diff
@@ -1,60 +1,26 @@
-import { CourseTaskDto } from 'api';
-import { StudentStats } from 'common/models';
+import { CourseScheduleItemDto, CourseScheduleItemDtoStatusEnum } from 'api';
 import dynamic from 'next/dynamic';
 import { useRouter } from 'next/router';
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { getQueryString } from 'utils/queryParams-utils';
 import CommonCard from './CommonDashboardCard';
 import { TasksStatsModal } from './TasksStatsModal';
 
 const TasksChart = dynamic(() => import('./TasksChart'), { ssr: false });
 
-export interface TasksStatistics {
-  completed: (CourseTaskDto | StudentStats)[];
-  notDone: (CourseTaskDto | StudentStats)[];
-  future: (CourseTaskDto | StudentStats)[];
-}
-
-export interface ITooltipItem {
-  datasetIndex: number;
-  index: number;
-  x: number;
-  xLabel: string;
-  y: number;
-  yLabel: number;
-}
-
-export interface ITooltipData {
-  datasets: IChartsConfigDataDatasets[];
-  labels: string[];
-}
-
-interface IChartsConfigDataDatasets {
-  data: number[];
-  backgroundColor?: string[];
-  hoverBackgroundColor?: string[];
-}
-
-enum GroupTaskName {
-  Completed = 'completed',
-  NotCompleted = 'notCompleted',
-  Future = 'future',
-}
+export type TaskStat = CourseScheduleItemDto & { comment?: string; githubPrUri?: string };
 
 type Props = {
-  tasks: TasksStatistics;
+  tasksByStatus: Record<CourseScheduleItemDtoStatusEnum, TaskStat[]>;
   courseName: string;
 };
 
 export function TasksStatsCard(props: Props) {
   const [isTasksStatsModalVisible, setTasksStatsModalVisible] = useState(false);
   const [statisticsTableName, setStatisticsTableName] = useState('');
-  const [selectedGroupTasks, setCurrentTaskStatsModal] = useState([] as (CourseTaskDto | StudentStats)[]);
+  const [selectedGroupTasks, setCurrentTaskStatsModal] = useState([] as TaskStat[]);
```

Comment:
Unclear naming. Please change it to `[selectedGropuTasks, setSelectedGroupTasks]` or `[currentTaskStatsModal, setCurrentTaskStatsModal]`

Reply:
removed them at all